### PR TITLE
Pass the transfer exception through as "previous"

### PR DESCRIPTION
**Pull request by jsandersuk**
- Provides access to server response to help identify errors

### CODE REVIEW

**Review by rccrdpccl — CHANGES_REQUESTED** [review]
Thanks for your contribution! Could you please add a small test with an example of the case we are intending to cover here?

**Comment by jsandersuk** [comment]
I have added a test to cover the suggested code change

**Review by rccrdpccl — APPROVED** [review]

**Comment by jsandersuk** [comment]
If you're happy with the changes could you please merge them into master at your earliest convenience?

Thanks for your help

File: src/Client.php [hunk 1]
```diff
@@ -32,7 +32,7 @@ public function query(string $query, array $variables = []): Response
         try {
             $response = $this->httpClient->request('POST', '', $options);
         } catch (TransferException $e) {
-            throw new \RuntimeException('Network Error.');
+            throw new \RuntimeException('Network Error.'.$e->getMessage(), null, $e);
```
Comment (joskfg) [inline src/Client.php]: The code is not right because the default is 0 not null, could you fix it? After this I'll merge the PR. Thanks!

**Review by joskfg — CHANGES_REQUESTED** [review]

**Comment by jsandersuk** [comment]
You are indeed correct; I have updated the code to be "0" rather than NULL

**Comment by rccrdpccl** [comment]
Thanks for your contribution!

**Pull request merged**